src/test/java/com/playground/playground/data/DataTest.java

```diff
@@ -0,0 +1,109 @@
+package com.playground.playground.data;
+
+import com.playground.playground.data.*;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Test class to validate the functionality of various components in the data package.
+ */
+public class DataTest {
+
+    /**
+     * Test the CircularDatasetGenerator to ensure it generates circular clusters correctly.
+     */
+    @Test
+    public void testCircularDatasetGenerator() {
+        DatasetGenerator circularDatasetGenerator = new CircularDatasetGenerator();
+        ArrayList<ArrayList<ArrayList<Double>>> circularDataset = circularDatasetGenerator.generate(10);
+
+        assertNotNull(circularDataset);
+        assertEquals(2, circularDataset.size());
+
+        ArrayList<ArrayList<Double>> cluster1 = circularDataset.get(0);
+        ArrayList<ArrayList<Double>> cluster2 = circularDataset.get(1);
+
+        assertFalse(cluster1.isEmpty());
+        assertFalse(cluster2.isEmpty());
+
+        assertTrue(cluster1.get(0).size() > 1000);
```

Comment:
So we are generating more than a 100 points maybe we could do an == so we can check exactly how many points are generated

Reply:
That sounds good, will implement that in the next PR.